app/controllers/api/v1/migrations/external_controller.rb

```diff
@@ -21,26 +21,41 @@ class ExternalController < ApiController
         end
 
         # POST /api/v1/migrations/roles.json
-        # Expects: { role: { :name } }
+        # Expects:
+        # { role: { :name,
+        #            role_permissions: { :CreateRoom, :CanRecord, :ManageUsers, :ManageRoles, :ManageRooms, :ManageRecordings, :ManageSiteSettings }}}
         # Returns: { data: Array[serializable objects(recordings)] , errors: Array[String] }
         # Does: Creates a role.
         def create_role
           role_hash = role_params.to_h
 
           return render_data status: :created if Role.exists? name: role_hash[:name], provider: 'greenlight'
 
-          role = Role.new role_hash.merge(provider: 'greenlight')
+          role = Role.new(name: role_hash[:name], provider: 'greenlight')
 
           return render_error status: :bad_request unless role.save
 
+          # Returns unless the Role has a RolePermission that differs from V3 default RolePermissions values
+          return render_data status: :created unless role_hash[:role_permissions].any?
+
+          # Finds & Updates the associated RolePermissions associated with the Role
+          role_permissions = RolePermission.where(role_id: role.id).includes(:permission)
+          # The role_permissions hash contains only the permissions with value that differs from V3 default values
+          role_hash[:role_permissions].any? && role_hash[:role_permissions].each do |name, value|
```

Comment:
Is the `.any?` necessary here? Can we not just do `.each`? I believe the `.each` can handle `nil`

Reply:
`role_hash[:role_permissions]` will return an empty hash, not `nil`
The same logic is repeated:
The hash will be populated with RolePermissions, RoomSettings that differs from V3 default values.
If no RolePermissions, RoomSettings differ from default values, `params_hash[:role_permissions]` and `params_hash[:room_settings] `will return an empty hash.